plaso/parsers/powershell_transcript.py

```diff
@@ -0,0 +1,343 @@
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      key (str): name of the parsed structure.
+      structure (pyparsing.ParseResults): structure of tokens derived from
+          a line of a text file.
+
+    Raises:
+      ParseError: when the structure type is unknown.
+    """
+    # the first line will always be a metadata line, however pyparsing grammar
+    # might misinterpret this since 'start-line' looks like a transcript line
+    if self.is_first_line and key != 'separator_line':
+      key = 'metadata_line'
+      self.is_first_line = False
+
+    # transcripts will begin/end at every second separator
+    # so we need to monitor for them
+    if self.found_first_separator is False and key == 'separator_line':
+      self.found_first_separator = True
+      self.key_structure.append((key, ' '.join(structure)))
+    # if 'second' separator is found, the current transcript
+    # is finished and we can create an event from it
+    elif self.found_first_separator is True and key == 'separator_line':
+      self.found_first_separator = False
+      # TODO: if the last transcript is not exited cleanly
+      # (i.e. there is no final separator), we miss the final transcript
+      # we would somehow need to know which line is the last one
+      # and call _CreateEventFromTranscript on it -- however I have not found
+      # a way to extract this information as a PyparsingMultiLineTextParser.
+      # pyparsing.StringEnd() did not match correctly
+      # __del__ will not be called in regular use cases
+      # tl;dr I currently have no idea how to fix this
+      self._CreateEventFromTranscript(parser_mediator)
+    else:
+      # overwrite key, since a transcript_line can never be followed directly
+      # by a metadata_line (and vice versa) and the pyparsing could be wrong
+      # (since a transcript_line can basically contain everything)
+      if (key == 'metadata_line' and len(self.key_structure) > 0
+        and self.key_structure[-1][0] == 'transcript_line'):
+        key = 'transcript_line'
+      elif (key == 'transcript_line' and len(self.key_structure) > 0
+        and self.key_structure[-1][0] == 'metadata_line'):
+        key = 'metadata_line'
+      self.key_structure.append((key, ': '.join(structure)))
+
+  def VerifyStructure(self, parser_mediator, lines):
+    """Verifies whether content corresponds to an PowerShell transcript file.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      lines (str): one or more lines from the text file.
+
+    Returns:
+      bool: True if this is the correct parser, False otherwise.
+    """
+    if lines[:25] == 'ï»¿' + self._SEPARATOR:
+      if 'Windows PowerShell' in lines:
+        return True
+    return False
+
+  def __del__(self):
```

Comment:
i would reconsider your design here -firstly please note this method is called when the object is about to be destroyed.  secondly, i'm not really sure why your unit tests would need to do this.

https://docs.python.org/3/reference/datamodel.html#object.__del__